### ISSUE TITLE
support SCRAM-SHA-1 auth

### ISSUE THREAD

**Issue by igreenfield** [allanbank/mongodb-async-driver#27]
Can you add support for 'SCRAM-SHA-1' auth?
without this auth with new mongo won't work.

**Comment by yairogen** [comment]
Yup, would like to see this as well.

**Comment by igreenfield** [comment]
I see this is in the trunk when you intend to release it??

**Comment by allanbank** [comment]
Yes we plan to release it in a formal release build but please realize that a release is actually a lot of work on our part. 

In addition to the automated functional tests that you see in the code based we also run endurance tests against clusters in various configurations (single node, replica set, sharded), using different authentication modes (none, mongodb auth, X509, LDAP, kerberos) and production versions (currently: 1.6.5, 1.8.5, 2.0.9, 2.2.X, 2.4.12, 2.6.12, 3.0.10, and 3.2.4). Most of that testing is done with YCSB and across Amazon clusters we setup and teardown so there is a monetary cost in addition to the labor/time costs.

Once the functional and endurance testing is completed we move to performance testing. That process can be quick or if it discovers a performance regression cause us to go back to the functional tests.

Is all of that testing strictly required, maybe not but I like knowing that we put some effort into validating the drivers we release but it means we are not going to release drivers every month or even every 6 months. 

As I said I hope to have an intern work on getting the new features pushed out this summer. Once that is complete we will push to get a release out.

As for the SCRAM-SHA-1 on master. Use at your own risk. We don't personally use it (prefer x.509) and I don't see the right automated authentication tests. I will try and get some time in the next week to validate the implementation and let you know here what I find.

**Comment by thiaguten** [comment]
Hello @allanbank , any news about this thread? I really like this driver is very performatic and well done. And I don't want use the official asynchronous driver version at all. 

I'm having this error:

```
[main] WARN  c.a.m.CredentialEditor - Unknown authMechanism: 'SCRAM-SHA-1'. Not authenticating. 
[main] WARN  c.a.m.c.c.a.AuthenticatingConnection - Authentication failed: [] 
com.allanbank.mongodb.error.MongoDbAuthenticationException: com.allanbank.mongodb.error.ReplyException: auth failed
```
